cli/js/lib.deno.shared_globals.d.ts

```diff
@@ -1347,7 +1374,63 @@ declare namespace performance {
    * console.log(`${t} ms since start!`);
    * ```
    */
-  export function now(): number;
+  now(): number;
+}
+
+declare const performance: Performance;
```

Comment:
The declaration in lib.dom.d.ts is slightly different:
```ts
/** Provides access to performance-related information for the current page. It's part of the High Resolution Time API, but is enhanced by the Performa
nce Timeline API, the Navigation Timing API, the User Timing API, and the Resource Timing API. */
interface Performance extends EventTarget {
    /** @deprecated */
    readonly navigation: PerformanceNavigation;
    onresourcetimingbufferfull: ((this: Performance, ev: Event) => any) | null;
    readonly timeOrigin: number;
    /** @deprecated */
    readonly timing: PerformanceTiming;
    clearMarks(markName?: string): void;
    clearMeasures(measureName?: string): void;
    clearResourceTimings(): void;
    getEntries(): PerformanceEntryList;
    getEntriesByName(name: string, type?: string): PerformanceEntryList;
    getEntriesByType(type: string): PerformanceEntryList;
    mark(markName: string): void;
    measure(measureName: string, startMark?: string, endMark?: string): void;
    now(): number;
    setResourceTimingBufferSize(maxSize: number): void;
    toJSON(): any;
    addEventListener<K extends keyof PerformanceEventMap>(type: K, listener: (this: Performance, ev: PerformanceEventMap[K]) => any, options?: boolean | AddEventListenerOptions): void;
    addEventListener(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListenerOptions): void;
    removeEventListener<K extends keyof PerformanceEventMap>(type: K, listener: (this: Performance, ev: PerformanceEventMap[K]) => any, options?: boolean | EventListenerOptions): void;
    removeEventListener(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | EventListenerOptions): void;
}
```

Reply:
Yes the `lib.don.d.ts` is aligned to what is GA in most browsers.  When I was comparing it to Chrome though, Chrome is implementing User Timing Level 3 (https://w3c.github.io/user-timing/) already, which is what this implementation is aligned to and has slightly better APIs.  It also doesn't implement the deprecated APIs that are marked in there and `setResourceTimingBufferSize` only applies to situations where you are keeping some internal resource timings (which we may want to do in the future).

@bartlomieju point above about it conflicting with the default `lib.dom.d.ts` is valid though and we should move them to the Deno NS lib.